hub/core/storage/local.py

```diff
@@ -0,0 +1,28 @@
+from hub.core.storage.provider_mapper import ProviderMapper
+import fsspec
+
+
+class LocalProvider(ProviderMapper):
+    """
+    Provider class for using the local filesystem.
+    """
+
+    def __init__(self, root: str):
+        """
+        Initializes the LocalProvider.
+
+        Example:
```

Comment:
I don't think these use-cases are complex enough to warrant examples in the docstring. Your call though